boot/bootutil/src/image_rsa.c

```diff
@@ -21,7 +21,7 @@
 
 #include "mcuboot_config/mcuboot_config.h"
 
-#ifdef MCUBOOT_SIGN_RSA
+#if defined(MCUBOOT_SIGN_RSA) || defined(MCUBOOT_SIGN_RSA3072)
```

Comment:
Looks like support for this is being added to bootutil before it's being added to mcuboot_config.h. Why? That wouldn't be the right order.

I also think there should be an `MCUBOOT_SIGN_RSA_LEN` (or `...MAX_LEN`) instead of a separate MCUBOOT_SIGN_RSA3072 option. It won't be long before 4096, after all (actually, I think that proving this design out to RSA 4096 as well would be a good idea to see how it will scale over the next few decades).

That would also allow you to set the size of `buf` below more accurately on smaller targets, conserving memory more carefully.

Reply:
I changed the order of the commits and now Zephyr/Mynewt config is added before the `bootutil` changes but I did not break for `mcuboot_config` specific commits. I also made the size of the key more parameterizable instead of having to create new ifdefery now to add RSA-4096 or some other size does not required `bootutil` changes (just OS specific configs).